### PR TITLE
Get ROCm building again on master

### PR DESCRIPTION
**Pull request by Jorghi12**
Big pile of fixes for ROCm build on master!

### CODE REVIEW

File: docker/caffe2/jenkins/common/install_rocm.sh [hunk 1]
```diff
@@ -2,8 +2,18 @@
 
 set -ex
 
+# TODO: This script should install a SPECIFIC ROCM_VERSION, but actually
+# it ignores all values of ROCM_VERSION which are not nightly.  Ugh!
 [ -n "$ROCM_VERSION" ]
 
+install_hip_nightly() {
```
Comment (bddppq) [inline docker/caffe2/jenkins/common/install_rocm.sh]: what does this install? will it override stuffs installed in install_ubuntu?

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -475,7 +475,7 @@ if(USE_ROCM AND NOT BUILD_CAFFE2)
  EXECUTE_PROCESS(COMMAND ${HIP_PATH}/bin/hipconfig --cpp_config OUTPUT_VARIABLE HIP_CXX_FLAGS)
 
  # Link with HIPCC https://github.com/ROCm-Developer-Tools/HIP/blob/master/docs/markdown/hip_porting_guide.md#linking-with-hipcc
- SET(CMAKE_CXX_LINK_EXECUTABLE ${HIP_HIPCC_EXECUTABLE})
+ # SET(CMAKE_CXX_LINK_EXECUTABLE ${HIP_HIPCC_EXECUTABLE})
```
Comment (ezyang) [inline cmake/Dependencies.cmake]: Next time, leave a comment saying what's going on with the commented out line

File: docker/caffe2/jenkins/common/install_rocm.sh (hunk 1 above)
Comment (ezyang) [reply docker/caffe2/jenkins/common/install_rocm.sh]: Yes, it builds and then installs some new deb files.

File: tools/amd_build/patches/a_aten_src_THC_generic_THCTensorRandom.cu.patch [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+index 906780b4f..b03e051cb 100644
 --- a/aten/src/THC/generic/THCTensorRandom.cu
 +++ b/aten/src/THC/generic/THCTensorRandom.cu
 @@ -504,11 +504,11 @@ THC_API void THCTensor_(clampedRandom)(THCState* state, THCTensor *self_, int64_
  #if defined(THC_REAL_IS_LONG) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_FLOAT)
    if (range > 1ULL << 32) {
      generate_random_64<<<NUM_BLOCKS, BLOCK_SIZE, 0, THCState_getCurrentStream(state)>>>(
 -        gen->state.gen_states, size, data, min_val, range);
-+        gen->state.gen_states, static_cast<int>(size), data, static_cast<std::conditional<std::is_same<real,float>::value, float, double>::type>(min_val), static_cast<std::conditional<std::is_same<real,float>::value, float, double>::type>(range));
++        gen->state.gen_states, static_cast<int>(size), data, min_val, range);
```
Comment (ezyang) [inline tools/amd_build/patches/a_aten_src_THC_generic_THCTensorRandom.cu.patch]: Let's try to get rid of this patch in the near future

**Review by ezyang — APPROVED** [review]

**Pull request merged**